fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2564,20 +2564,27 @@ private Bundle createSearchBundle(List<Resource> resources, FHIRSearchContext se
             }
 
             for (Resource resource : resources) {
-                if (resource.getId() == null) {
-                    throw new IllegalStateException("Returned resources must have an id.");
+                Bundle.Entry.Builder entryBuilder = Bundle.Entry.builder();
```

Comment:
Are there any unit tests covering this? Getting null data for testing probably requires some dedicated test DAO to forcibly set the data value to null.

Reply:
No, not specifically. I could, but wasn't sure it was worth the overhead, etc.

Reply:
I added a testcase to FHIRRestHelperTest.java to exercise this code path.